test/firebase_auth_mocks_test.dart

```diff
@@ -239,15 +239,46 @@ void main() {
     expect(idTokenResult.claims, {'weight': 70});
   });
 
-  test('Returns null after sign out', () async {
-    final auth = MockFirebaseAuth(signedIn: true, mockUser: tUser);
-    final user = auth.currentUser;
+  group('Sign out', () {
+    test('Returns null after sign out', () async {
+      final auth = MockFirebaseAuth(signedIn: true, mockUser: tUser);
+      final user = auth.currentUser;
 
-    await auth.signOut();
+      await auth.signOut();
+
+      expect(auth.currentUser, isNull);
+      expect(auth.authStateChanges(), emitsInOrder([user, null]));
+      expect(auth.userChanges(), emitsInOrder([user, null]));
+    });
+
+    test('Can sign in again after sign out', () async {
+      final auth = MockFirebaseAuth(signedIn: true, mockUser: tUser);
+      final user = auth.currentUser;
+
+      await auth.signOut();
 
-    expect(auth.currentUser, isNull);
-    expect(auth.authStateChanges(), emitsInOrder([user, null]));
-    expect(auth.userChanges(), emitsInOrder([user, null]));
+      auth.mockUser = tUser;
```

Comment:
Since signing out does not set `mockUser` to `null` anymore, do you still need to set `mockUser` again if signing in with the same user? Maybe you can remove this line.

Reply:
Ah yes, you're right - this should have been removed too. I'll clean that up

Reply:
Ah, looks like you fixed it - I missed your other comment. Thanks!